Using local middleware in a blueprint

Hello,
I am trying to use local middleware in a blueprint.

I defined my application as follows:
```cpp
crow::App<crow::CORSHandler, AuthorisationMiddleware> app;
```
Then I define my blueprint
```cpp
crow::Blueprint api = initApi(app);
app.register_blueprint(api);
```
`initApi()` function contains:
```cpp
crow::Blueprint initApi(crow::App<crow::CORSHandler, AuthorisationMiddleware> &app)
{
	crow::Blueprint api("api");
	CROW_BP_ROUTE(api, "/users/<string>")
		.methods(crow::HTTPMethod::Delete)
		.CROW_MIDDLEWARES(app, AuthorisationMiddleware)
		([&](const crow::request& req, crow::response& res, std::string username)
			{
				// Handler continues here
				res.end();
				return;
			});

	return api;
}
```

But when I try to compile, I am getting the following errors:
```
error C2672: 'operator __surrogate_func': no matching overloaded function found
error C2783: 'void crow::detail::handler_call_bridge<crow::TaggedRule<crow::black_magic::single_tag_to_type<4>::type>,crow::Crow<crow::CORSHandler,AuthorisationMiddleware> &,AuthorisationMiddleware>::operator ()(F &&)': could not deduce template argument for 'F'
```

--- a/include/crow/app.h
+++ b/include/crow/app.h
@@ -29,7 +29,7 @@
 #else
 #define CROW_ROUTE(app, url) app.route<crow::black_magic::get_parameter_tag(url)>(url)
 #define CROW_BP_ROUTE(blueprint, url) blueprint.new_rule_tagged<crow::black_magic::get_parameter_tag(url)>(url)
-#define CROW_MIDDLEWARES(app, ...) middlewares<decltype(app), __VA_ARGS__>()
+#define CROW_MIDDLEWARES(app, ...) middlewares<std::remove_reference<decltype(app)>::type, __VA_ARGS__>()
 #endif
 #define CROW_CATCHALL_ROUTE(app) app.catchall_route()
 #define CROW_BP_CATCHALL_ROUTE(blueprint) blueprint.catchall_rule()

